### ISSUE TITLE
Suggestions on BOM, PCB layout

### ISSUE THREAD

**Issue by Basilisvirus** [ataradov/usb-sniffer#51]
I will replicate this open source project soon. I am looking at the files right now and I would like a feedback from the suggestions Im making :) 

1. For easier hand-soldering, without the need of a hot air station I suggest IC4 to have a large via hole in the middle of it:  
![image](https://github.com/user-attachments/assets/ce68d3e2-1ba3-4d37-b55b-7cac71a47e5e)  

This way the middle pad can be soldered from the bottom side by filling it with solder.   
Be careful: make sure you slowly pour solder in the hole (start pouring solder to the side of the hole and wait for it get hot enough so that the solder will slow in on its own) and that the solder has "spourted" inside the hole on its own, to make sure the solder actually touched the pad and it didnt just cover the hole.
 
2. For mouser.com, IC6 (DSC6101CI2A-026.0000) is obsolete:  
![{C6AAD7E0-9A09-434C-BA5E-34476755A2AF}](https://github.com/user-attachments/assets/76dbc089-3477-4851-aa41-956a70de2aa3)

I think DSC6003MI2B-026.0000 is a good replacement. It is also smaller in size (VFLGA-4 footprint) , which means less load capacitance, easier to solder on the pre-existing pad without how air since the SMD pads now stick out.:  
![{E38C46AD-57F7-46F3-A847-693126AB8C82}](https://github.com/user-attachments/assets/2e5d6ccb-bb77-45cc-a418-b80a94965457)

3. Same case for DSC6101CI2A-024.0000:  
![{2E266DAF-A9C8-4B25-BA96-E144D2DFF64C}](https://github.com/user-attachments/assets/3d236773-1f7a-4e8d-8e7d-40562bd5a475)
The DSC6001HI2B-024.0000 (VFLGA-4 footprint as well)  looks like a good replacement. Its smaller and the pads (again) stick out, so easier to solder with hand soldering iron.

4. This doesnt matter really, for the frequencies and length of the traces, but you will have impedance mismatch when the copper changes width, to the places where I point the arrows. A solution to that is to make the width not change so much, aka make zones around the pads:  
![image](https://github.com/user-attachments/assets/03bfce0a-9bea-4f76-abea-acd6d8e428ff)  
This happens in some other places on the board.  

5. This GND copper does not have vias close to it. So it might as well act as a floating copper and is a conductor for noise to get from one trace to another. Two solutions here. Either let enough space for GND vias or place the coppers close enough so that no GND copper will get there.  
![image](https://github.com/user-attachments/assets/f86f9c8a-05bb-4bb4-9c80-c0d92205b2c6)  
I applied the second fix for example:  
![{6526BC47-82F3-42FB-BD6E-7A171B3DBD05}](https://github.com/user-attachments/assets/e8c5050e-0026-4b60-9e6e-3cfd99b7b338)

6. I didnt see any instruction to order the PCB in 1mm instead of 1.55mm FR4. I guess 1mm will behave better, as the two layers will be closer. For that reason, I am mainly concerned about USB impedance. based on your USB D+/D- distance and trace thickness, the impedance of the USB differential data lines is 136.64Ω on 1mm PCB, this increases to 147Ω if 1.55mm FR4 is used.
![image](https://github.com/user-attachments/assets/edd5d40b-fd8e-4b66-82ef-fadd8ad66290)  

(I will add new suggestions as I check more stuff. Let me know if I am wrong somewhere)

**Comment by ataradov** [comment]
The hole is up to you. The IC is small and reasonably easy to solder by hand as it is.
None of the PCB stuff really matters, you can do it if it makes you sleep better.
And for the crystals, there is already multiple discussions with replacements from LCSD. The footprint universal, there are dozens of possible alternatives.

**Comment by stappersg** [comment]
On Thu, Dec 26, 2024 at 07:28:56AM -0800, Christianidis Vasileios wrote:
> Closed #51 as completed.

For the PCB-layout, Yes.

For the Bill Of Materials change, hope I to see a merge request.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by ataradov** [comment]
I'm not going to accept merge requests. I need this repo to  match what I have. Different people will have access to the different parts and it is impossible to accommodate everyone. Plus things change over time. If you are building this, substitutions like that should be easy anyway.

**Comment by Basilisvirus** [comment]
Agree with ataradov. Since he is so active on the issues, if you need any clarification about the BOM, one might as well raise an issue for it. Changing BOM will mean that ataradov will need to re-solder stuff and change things. This takes time and effort. 

If anyone manages anything better they can make an issue, fork the project on their github and make the desired changes.
